docs/managed-tests/group-tests.md

```diff
@@ -0,0 +1,138 @@
+# Group Tests
+
+Group tests allow you to organize and execute multiple test runs together, either as a logical group or as a batch. This feature is particularly useful when you need to run multiple tests in sequence or want to organize related tests together.
+
+## Overview
+
+Group tests work with all test types, but there's an important distinction in how they're executed:
+
+- **Local Tests (E2E and Activation):** Run on your local machine using the QIT CLI.
+- **Remote Tests (All other test types):** Run on QIT's infrastructure.
+
+## Basic workflow
+
+1. **Add tests to a group** using the `--group` flag
+2. **Register the group** (optional) with a custom identifier
+3. **Run the group** to execute all tests
+
+## Adding tests to a group
+
+To add a test to a group, simply append the `--group` flag to any test command:
+
+```qitbash
+qit run:activation your-extension --group
+```
+
+You can add multiple tests to the same group:
+
+```qitbash
+qit run:activation automatewoo --group --extension_set compatibility
+qit run:security your-extension --group
+```
+
+## Registering a group
+
+While optional, registering a group with a custom identifier helps you track and reference it later:
+
+```qitbash
+qit group:register --group-identifier commit-1234567890
+```
+
+## Running a group
+
+You can run a group in two ways:
+
+1. **Run without registration:**
+```qitbash
+qit group:run
+```
+
+2. **Run with a specific group identifier:**
+```qitbash
+qit group:run --group-identifier commit-1234567890
+```
+
+When you run a group:
+- Remote tests are triggered on QIT's infrastructure
+- Local tests (E2E and Activation) run sequentially on your machine
+- No additional input is required once the group starts running
+
+## Batch mode
+
+If you want to run tests without grouping them logically, use the `--skip-grouping` flag:
+
+```qitbash
+qit group:run --skip-grouping
+```
+
+This will execute the tests without maintaining the group relationship between them.
+
+## Managing groups
+
+### View current group
+
+To see the currently stored group:
+
+```qitbash
+qit group:show
+```
+
+### Fetch a specific group
+
+To retrieve a specific group from remote:
+
+```qitbash
+qit group:fetch --group-identifier commit-1234567890
+```
+
+### Clear the current group
+
+To remove the current group:
+
+```qitbash
+qit group:clear
+```
+
+## Important notes
+
+- **Duplicate prevention:** If a test with identical parameters is already in the group, it won't be added again
+- **Opting out:** To prevent a test from being picked up by a group run, use the `--no_group` flag
+- **Group status:** You can monitor the status of your group and individual test runs through the QIT manager interface
```

Comment:
Might be worth clarifying what "QIT manager" interface means here; if it's the backend only we may want to drop this line.

Reply:
Thanks for pointing that out. I'm removing that bullet along with the `--no_group` bullet as well.